sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -408,7 +408,13 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
   }
 
   private def defaultTablePath(tableIdent: TableIdentifier): String = {
-    val dbLocation = getDatabase(tableIdent.database.get).locationUri
+    // default database's location always use the warehouse path,
+    // and since the location of database stored in metastore is qualified,
+    // here we also make qualify for warehouse location
+    val dbLocation = if (tableIdent.database.orNull == SessionCatalog.DEFAULT_DATABASE) {
+      SessionCatalog.makeQualifiedPath(conf.get(WAREHOUSE_PATH), hadoopConf).toString
+    } else getDatabase(tableIdent.database.get).locationUri
```

Comment:
yes, it is duplicate, I have tried to merge them by unify the `defaultTablePath for managed table` in [HiveExternalCatalog](https://github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala#L205-L216) and [InMemoryCatalog](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala#L194-L202
) up to [SessionCatalog](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala#L253)

then, we can drop the `defaultTablePath` in `HiveExternalCatalog`.

But this cause some test faile in `InMemoryCatalogSuit`, we should modify the tests.

So I think maybe we can keep the already existed duplication function `defaultTablePath`, and to merge them and fix some test failed in another PR.